### ISSUE TITLE
Bump okta.sdk.version from 8.2.3 to 10.x.x

### ISSUE THREAD

**Issue by jfb-NRB** [okta/okta-spring-boot#569]
### I'm submitting a
  - [ ] bug report
  - [X] feature request

### Background info

Can you upgrade the project to new stable version of the java sdk 10.x.x because the version 8.x.x have the status Retired.

### Expected behavior

Use the new 10.x.x stable version of the sdk.

### SDK Version

2.1.7 and 3.0.3

**Comment by woemler** [comment]
Are there any compatibility issues with using the SDK version 10+ with the existing `okta-spring-boot` releases?

**Comment by arvindkrishnakumar-okta** [comment]
@woemler yes, there are compatibility issues due to the several major breaking changes introduced in okta-sdk-java from 8.x to 1.x+.

**Comment by jfb-NRB** [comment]
How soon could the upgrade be done? The last release sdk version is 14 now.

**Comment by arvindkrishnakumar-okta** [comment]
I have a [PR](https://github.com/okta/okta-spring-boot/pull/617) (draft) in the making, will get it merged soon after resolving few issues around it. Stay tuned!

**Comment by lealceldeiro** [comment]
are there any plans to get that pr merged?
